include/lsst/afw/detection/Footprint.h

```diff
@@ -329,6 +329,25 @@ class Footprint : public afw::table::io::PersistableFacade<lsst::afw::detection:
      */
     void removeOrphanPeaks();
 
+    /**
+     * Compute the significance of each peak, given a single value for sigma.
+     *
+     * @param sigma The standard deviation for peak detection, positive or negative as appropriate.
+     *
+     * @throws pex::exceptions::NotFoundError Thrown if the the peaks do not have a significance field.
+     */
+    void computePeakSignificance(double sigma);
+
+    /**
+     * Compute the significance of each peak, using the variance image.
+     *
+     * @param variance The variance plane of the image that this footprint is in.
+     * @param polarity +1 if this is detected as a positive footprint, -1 if negative.
+     *
+     * @throws pex::exceptions::NotFoundError Thrown if the the peaks do not have a significance field.
+     */
+    void computePeakSignificance(image::Image<float> const &variance, int polarity);
```

Comment:
Should document that these put the significance values in a peak field named "significance", which must already be part of the Footprint's peak schema.  I'd also be inclined to name them "updatePeakSignificance" instead of "computePeakSignificance", because I think that better captures the modify-in-place behavior, but that's really splitting hairs.

Reply:
I wasn't sure about the name, and I like `updatePeakSignificance` better. Thanks.